apis/v1alpha1/route_types.go

```diff
@@ -115,18 +115,20 @@ type RouteForwardTo struct {
 	Port PortNumber `json:"port"`
 
 	// Weight specifies the proportion of traffic forwarded to the backend
-	// referenced by the ServiceName or BackendRef field. computed as
+	// referenced by the ServiceName or BackendRef field. This is computed as
 	// weight/(sum of all weights in this ForwardTo list). Weight is not a
-	// percentage and the sum of weights does not need to equal 100. If only one
-	// backend is specified, 100% of the traffic is forwarded to that backend.
-	// If weight is set to 0, no traffic should be forwarded for this entry.
-	// If unspecified, weight defaults to 1.
+	// percentage and the sum of weights does not need to equal 100.
+	//
+	// If only one backend is specified and it has a weight greater than 0, 100%
+	// of the traffic is forwarded to that backend. If weight is set to 0, no
+	// traffic should be forwarded for this entry. If unspecified, weight
+	// defaults to 1.
 	//
 	// Support: Extended
 	//
 	// +kubebuilder:default=1
 	// +kubebuilder:validation:Minimum=0
-	// +kubebuilder:validation:Maximum=10000
+	// +kubebuilder:validation:Maximum=1000000000
 	Weight int32 `json:"weight,omitempty"`
```

Comment:
Probably scope-creep for this PR: We should DRY this out and have a `type Weight int32` in this package.

Reply:
https://github.com/kubernetes-sigs/service-apis/issues/440